### PR TITLE
Improve example documentation

### PR DESCRIPTION
**Pull request by mharris-dstl**
Should improve readability and ensure flake8 tests pass for some examples. Just documentation addition - no coding changes.

Have also changed types/association to fix a typo raised by @A-acuto and put in some full stops while I was there.

### CODE REVIEW

File: docs/examples/Track_Stitcher_Example.py [hunk 1]
```diff
@@ -290,12 +279,11 @@
 # ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 # Now that we have a set of tracklets, we can apply the Track Stitching method to stitch tracklets
 # together into tracks. The code in the following cell applies this process using the class
-# `TrackStitcher` and plots the stitched tracks.
-# `TrackStitcher` has a property search_window that enables you to reduce compute time by filtering
-# out track segments that are not within a determined time window. When forward stitching, the
-# associator will consider any track that has a start point that falls within the time window
-# :math:`(t, t + search_window)`. When backward stitching, the associator will consider tracks that
-# have an endpoint within the time window :math:`(t - search_window, t)`.
+# `TrackStitcher` and plots the stitched tracks. `TrackStitcher` has a property 'search_window' that
```
Comment (orosoman-dstl) [inline docs/examples/Track_Stitcher_Example.py:282]:
```suggestion
# :class:`~.`TrackStitcher` and plots the stitched tracks. :class:`~.`TrackStitcher` has a property ``search_window`` that
```
Enclose code in double \`\`code_here\`\`, which is likely below the Esc key at the top left of your keyboard. I'll let you find all the places this needs fixing.

File: docs/examples/Track_Stitcher_Example.py [hunk 2]
```diff
@@ -290,12 +279,11 @@
...
+# reduces computation time by filtering out track segments that outside a defined time window. When
+# forward stitching, the associator will consider any track that has a start point that falls within
+# the time window :math:`(t, t + search_window)`. When backward stitching, the associator will
```
Comment (orosoman-dstl) [inline docs/examples/Track_Stitcher_Example.py:285]:
```suggestion
# the time window :math:`(t, t + search\_window)`. When backward stitching, the associator will
```
You need to escape the underscore (I think with `\`) or the W in "window" is made subscript.

File: docs/examples/Track_Stitcher_Example.py [hunk 3]
```diff
@@ -13,9 +13,9 @@
...
 # by considering the state of a tracked object and predicting its state at a future (or past) time.
-# This example generates a set of `tracklets`, before applying track stitching. The figure below
-# visualizes the aim of track stitching: taking a set of tracklets (left, black) and producing a set
-# of tracks (right, blue/red).
+# This example generates a set of `tracklets` and applies track stitching to them. The figure below
+# visualizes the aim of track stitching: taking a set of tracklets (left, black) and producing a
+# set of tracks (right, blue/red).
 
 # %%
 # .. image:: ../_static/track_stitching_basic_example.png
```
Comment (orosoman-dstl) [inline docs/examples/Track_Stitcher_Example.py:21]: Is it possible to make the images a bit larger? They look quite small.

File: docs/examples/Track_Stitcher_Example.py [hunk 4]
```diff
@@ -25,30 +25,29 @@
...
 # ^^^^^^^^^^^^^^^^^^
-# For a given track section, we consider the state at the end-point of the track, say state
-# :math:`x` at the time that the observation was made, call this time :math:`k`. We use the state of
-# the object to predict the state at time :math:`k + \delta k`. If the state at the start point of
-# another track section falls within an acceptable range of this prediction, we may associate the
+# For a given track section, consider the state at the end-point of the track :math:`x` at
+# the time :math:`k` that the observation was made. We use the state of the object to predict its
+# state at time :math:`k + \delta k`. If the state at the start point of
+# another track section falls within an acceptable range of this prediction, we associate the
 # tracks and stitch them together. This method is used in the function `forward_predict`.
```
Comment (orosoman-dstl) [inline docs/examples/Track_Stitcher_Example.py:39]:
```suggestion
# tracks and stitch them together. This method is used in the function ``forward_predict``.
```

**Review by orosoman-dstl — CHANGES_REQUESTED** [review]
Looking good, just be careful to enclose variable names etc. in double \`\` (see https://devguide.python.org/documentation/markup/) when referring to them inline, and a few other minor comments.

**Comment by A-acuto** [comment]
I've just seen a small typo (missing colon before class:~) in the association description of the function `associations_including_objects` in the [AssociationSet](https://github.com/dstl/Stone-Soup/blob/improve_example_documentation/stonesoup/types/association.py) data type , if you prefer I can fix it here in this PR

**Comment by A-acuto** [comment]
Hi, I've found a documentation typo in `functions.build_rotation_matrix`, likely an extra tab in the first entry of the Parameters part which breaks the layout. Like this: 
image

Probably removing the tab before `angle_vector`, and keeping the rest of the text as it is, will fix this issue.

File: docs/examples/Track_Stitcher_Example.py (hunk 3 above)
Comment (mharris-dstl) [reply docs/examples/Track_Stitcher_Example.py:21]: Yes but in the interest of the size of this PR, I don't want to do that here. Would rather get the current changes through first, and can make other changes later on :)

**Comment by mharris-dstl** [comment]
> Hi, I've found a documentation typo in `functions.build_rotation_matrix`, likely an extra tab in the first entry of the Parameters part which breaks the layout. Like this: image
> 
> Probably removing the tab before `angle_vector`, and keeping the rest of the text as it is, will fix this issue.

Happy to do this in a follow-up PR - just don't want to keep making this one bigger when it's primarily about example docs :)

File: docs/examples/Sensor_Platform_Simulation.py [hunk 5]
```diff
@@ -1,23 +1,25 @@
...
+capability.
 """
 
 # %%
 # Building a Simulated Sensor Platform
 # ------------------------------------
-# The focus of this example is to show how to setup and configure simulations, as such the application of a tracker
-# will not be covered in detail. For more information about trackers and how to configure them review of the
-# tutorials and demonstrations is recommended.
+# The focus of this example is to show how to set up and configure simulations. as such, the
```
Comment (hpritchett-dstl) [inline docs/examples/Sensor_Platform_Simulation.py:13]:
```suggestion
# The focus of this example is to show how to set up and configure simulations. As such, the
```

File: docs/examples/Sensor_Platform_Simulation.py [hunk 6]
```diff
@@ -146,63 +152,73 @@
...
 platform.sensors
 
 # %%
-# You will notice that platform.sensors returns a list which contains our single sensor. This hints at the multi-sensor
-# platform functionality which is shown in a subsequent example.
+# You will notice that ``platform.sensors`` returns a list which contains our single sensor.
+# This hints at the multi-sensor platform functionality which is shown in a subsequent example.
 #
-# We can also check to ensure that the default mounting_offsets have been applied:
+# We can also check that the default mounting_offsets have been applied:
```
Comment (hpritchett-dstl) [inline docs/examples/Sensor_Platform_Simulation.py:182]:
```suggestion
# We can also check that the default mounting offsets have been applied:
```

File: docs/examples/Sensor_Platform_Simulation.py [hunk 7]
```diff
@@ -146,63 +152,73 @@
...
 radar.mounting_offset
 
 # %%
-# And that the rotation_offsets have been applied:
+# and that the rotation_offsets have been applied:
```
Comment (hpritchett-dstl) [inline docs/examples/Sensor_Platform_Simulation.py:186]:
```suggestion
# and that the rotation offsets have been applied:
```

**Comment by mharris-dstl** [comment]
Have committed changes made by @hpritchett-dstl

**Review by hpritchett-dstl — APPROVED** [review]

File: docs/examples/Sensor_Platform_Simulation.py [hunk 8]
```diff
@@ -133,8 +138,9 @@
 noise_covar = CovarianceMatrix(np.array(np.diag([np.deg2rad(3)**2,
                                                  np.deg2rad(0.15)**2,
                                                  25**2])))
-# this radar measures range with an accuracy of +/- 25m, and elevation accuracy +/- 3
-# degrees and bearing accuracy of +/- 0.15 degrees
+
+# this radar measures range with an accuracy of +/- 25m,elevation accuracy +/- 3
```
Comment (csherman-dstl) [inline docs/examples/Sensor_Platform_Simulation.py:142]:
```suggestion
# this radar measures range with an accuracy of +/- 25m, elevation accuracy +/- 3
```

File: docs/examples/Disjoint_Tracking_Classification.py [hunk 9]
```diff
@@ -364,13 +363,13 @@ def measurement_model(self):
 # The hypothesiser is a :class:`~.CompositeHypothesiser` type. It is in the data association step
 # that tracking and classification are combined: for each measurement, a hypothesis is created for
 # both a track's kinematic and categorical components. A :class:`~.CompositeHypothesis` type is
-# created, which contains these sub-hypotheses, whereby its weight is equal to the product of the
-# sub-hypotheses' weights. These sub-hypotheses should be probabilistic.
+# created which contains these sub-hypotheses, whereby its weight is equal to the product of the
+# sub-hypotheses weights. These sub-hypotheses should be probabilistic.
```
Comment (csherman-dstl) [inline docs/examples/Disjoint_Tracking_Classification.py:367]: unsure but the weight is the possession of the sub-hypotheses

```suggestion
# sub-hypotheses' weights. These sub-hypotheses should be probabilistic.
```

File: docs/examples/Track_Stitcher_Example.py [hunk 10]
```diff
@@ -277,25 +266,26 @@
...
-# together into tracks. The code in the following cell applies this process using the class
-# `TrackStitcher` and plots the stitched tracks.
-# `TrackStitcher` has a property search_window that enables you to reduce compute time by filtering
-# out track segments that are not within a determined time window. When forward stitching, the
-# associator will consider any track that has a start point that falls within the time window
-# :math:`(t, t + search_window)`. When backward stitching, the associator will consider tracks that
-# have an endpoint within the time window :math:`(t - search_window, t)`.
+# together into tracks. The code in the following cell applies this process using the
+# :class:`~.TrackStitcher` and plots the stitched tracks. :class:`~.TrackStitcher` has a property
+# 'search_window' that reduces computation time by filtering out track segments that outside a
```
Comment (csherman-dstl) [inline docs/examples/Track_Stitcher_Example.py:284]:
```suggestion
# 'search_window' that reduces computation time by filtering out track segments that are outside a
```

**Review by csherman-dstl — COMMENTED** [review]
Added a few minor grammatical comments

**Review by csherman-dstl — APPROVED** [review]

**Pull request merged**